### ISSUE TITLE
$ in JSON path multiple levels deep returns empty object

### ISSUE THREAD

**Issue by rmtuckerphx** [perpk/json-xform#37]
Here is the code sample:
```
const jsonXform = require("@perpk/json-xform")
const source = {
    "$data": {
        "fullName": "Mark"
    }
};

const mapping = {
    "fieldset": [
        {
            "from": "$data.fullName",
            "to": "name"
        }

    ]
};

// valid JSON
//source.$data.fullName
//source['$data'].fullName
//source['$data']['fullName']

jsonXform.mapToNewObject(source, mapping)
```

When I include a source that has a path that include a special character and the path is multiple levels deep, the mapping doesn't work.

The result is:
```
{}
```

Expected result:
```
{name: "Mark"}
```

I have a project where json-xform would be a perfect fit when we can get past the special chars in path issues. The depth of the source JSON could be many levels deep. And multiple of those levels could have special chars in the name.

Thanks you!

**Comment by perpk** [comment]
@rmtuckerphx thanks for spotting this one. I'll have a look asap!

**Comment by perpk** [comment]
@rmtuckerphx I took care of the issue and released with version 1.0.3.
According to the fix special characters should now be supported across all levels of nesting.

Thanks a lot!
Cheers!

**Comment by rmtuckerphx** [comment]
@perpk  I thought you might want to see this Twitter thread where I am using json-xform for my implementation of Data Transforms.
https://twitter.com/marktucker/status/1514262208330420225
